src/main/java/org/opensearch/sdk/handlers/OpensearchRequestHandler.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk.handlers;
+
+import org.opensearch.extensions.OpenSearchRequest;
+import org.opensearch.sdk.NamedWriteableRegistryAPI;
+import org.opensearch.sdk.ExtensionsRunner;
+import org.opensearch.transport.TransportResponse;
+
+/**
+ * This class handles the request from OpenSearch to a {@link ExtensionsRunner#startTransportService(TransportService transportService)} call.
+ */
+
+public class OpensearchRequestHandler {
+    private static NamedWriteableRegistryAPI namedWriteableRegistryApi = new NamedWriteableRegistryAPI();
+
+    /**
+     * Handles a request from OpenSearch and invokes the extension point API corresponding with the request type
+     *
+     * @param request  The request to handle.
+     * @return A response to OpenSearch for the corresponding API
+     * @throws Exception if the corresponding handler for the request is not present
+     */
+    public static TransportResponse handleOpenSearchRequest(OpenSearchRequest request) throws Exception {
+        // Read enum
+        switch (request.getRequestType()) {
+            case REQUEST_OPENSEARCH_NAMED_WRITEABLE_REGISTRY:
+                return namedWriteableRegistryApi.handleNamedWriteableRegistryRequest(request);
+            // Add additional request handlers here
+            default:
+                throw new Exception("Handler not present for the provided request");
```

Comment:
One of my pet peeves is broad Exceptions.  Pick a subclass.   IllegalArgumentException is probably a good choice.  Also add what was wrong, e.g., show the type that isn't handled.

Reply:
I will change that in next commit.